source/patterns/@aws-solutions-constructs/resources/lib/template-writer-custom-resource/index.mjs

```diff
@@ -0,0 +1,73 @@
+/**
+ *  Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License"). You may not use this file except in compliance
+ *  with the License. A copy of the License is located at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  or in the 'license' file accompanying this file. This file is distributed on an 'AS IS' BASIS, WITHOUT WARRANTIES
+ *  OR CONDITIONS OF ANY KIND, express or implied. See the License for the specific language governing permissions
+ *  and limitations under the License.
+ */
+
+import { S3Client, GetObjectCommand, PutObjectCommand } from "@aws-sdk/client-s3";
+import * as crypto from 'crypto';
+
+const s3Client = new S3Client({ region: process.env.REGION });
+
+export const handler = async (event, context) => {
+  // tslint:disable-next-line:no-console
+  console.log(`Received: ${JSON.stringify(event, null, 2)}`);
+
+  let status = 'SUCCESS';
+  let responseData = {};
+
+  if (event.RequestType === 'Create' || event.RequestType === 'Update') {
+    try {
+      const templateValues = event.ResourceProperties.TemplateValues;
+      const templateInputBucket = event.ResourceProperties.TemplateInputBucket;
+      const templateInputKey = event.ResourceProperties.TemplateInputKey;
+      const templateOutputBucket = event.ResourceProperties.TemplateOutputBucket;
+      const templateOutputKey = crypto.randomBytes(32).toString('hex');
```

Comment:
the first three lines are standard in getting custom information into the custom resource, and then the 4th line that generates the random key is good here because we avoid having to pull in a third party dependency like `uuid`

Reply:
addressed offline